azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java

```diff
@@ -71,6 +73,10 @@ private void setupAllMetrics() {
     this.queueWaitMeter = this.metricsManager.addHistogram(QUEUE_WAIT_HISTOGRAM_NAME);
   }
 
+  public <T> void addSessionCount(final Supplier<T> getSessionCount) {
+    this.metricsManager.addGauge(SESSION_COUNT, getSessionCount);
```

Comment:
Have you thought about registering this metric after SessionCache's initialization, outside its constructor? Maybe SessionCache doesn't need to know about metrics existence.